### ISSUE TITLE
Feature Request: In-game Chat stream to Discord Text Channel

### ISSUE THREAD

**Issue by PitchBright** [CannibalVox/McDiscord#4]
Would it be possible to have the in-game chat stream to a Discord server's text channel? MC to Discord, and Discord to MC… like IRC allows.

This would be configurable, so the server admin can define the name of the text channel, in the mcdiscord.cfg

**Comment by CannibalVox** [comment]
Eventually, but there's no good way to allow this yet.  There will be a large-scale API for discord in the future and then we'll add those features.

**Comment by PitchBright** [comment]
Awesome. I figured it was premature (to be asking for it), but I wanted to get it down on paper for ya, so the idea didn't get lost for down the road.

**Comment by canilsen** [comment]
Would it be possible to have a Bot pull chat messages from a log written by mcdiscord?

**Comment by awade33** [comment]
Is there any update on this? I think this is really the most needed feature as the mod doesn't really do much atm so there's no real need to use it.

**Comment by tomaste** [comment]
Agreed with Awade33, I'm looking into setting up a MC->IRC->discord solution and it looks hairy.  Would be so nice to just go straight from MC->Discord.

**Comment by awade33** [comment]
If you get anything working, tomaste, please let me know :)

**Comment by tomaste** [comment]
So here's a possible high level scenario:
1. Find somewhere to host an IRC channel (QuakeNET, Gamesurge, etc.)
2. Register an Admin nick (the network will have a page on how to do that.
3. Register a channel as the admin nick.
4. Register a nick for each MC server that will join
5. Set them to Auto +v (voice) or +o (Operator) in the IRC Channel above.
6. Add Purple IRC (http://minecraft.curseforge.com/projects/purpleirc) to each server. (forge and bukkit versions available)
7. host and run this IRC->discord bridge:  https://github.com/reactiflux/discord-irc

I think I can manage all the way up to step 7, but I've never done anything with Linux, so that is where my unknown territory begins.

**Comment by awade33** [comment]
I've played with Linux a bit. I've got some time tomorrow so I'll try what
you've said and get back to you.

**Comment by TehSeph** [comment]
https://github.com/DV8FromTheWorld/JDA/
This help any? It's the most feature-complete Java Discord library.

[comparison source: https://discordapi.com/unofficial/comparison.html]

**Comment by canilsen** [comment]
use this instead: http://minecraft.curseforge.com/projects/discordchat

**Comment by TehSeph** [comment]
@canilsen That's using JDA (the library I linked) lol https://github.com/shadowfacts/DiscordChat

**Comment by canilsen** [comment]
Exactly. Use that mod instead, since this mod is dead
